easybuild/tools/module_generator.py

```diff
@@ -298,6 +299,100 @@ def use(self, paths, prefix=None, guarded=False):
         """
         raise NotImplementedError
 
+    def _generate_help_text(self):
+        """
+        Generate syntax-independent help text used for `module help`.
+        """
+
+        def generate_section(sec_name, sec_txt, strip=False):
+            """
+            Generate section with given name and contents.
+            """
+            if sec_txt:
+                if strip:
+                    text = sec_txt.strip()
+                else:
+                    text = sec_txt
+
+                return [
+                    '',
+                    '',
+                    sec_name,
+                    '=' * len(sec_name),
+                    text,
+                ]
+            else:
+                return []
+
+        # General package description (mandatory)
+        lines = generate_section('Description', self.app.cfg['description'], strip=True)
+
+        # Package usage instructions (optional)
+        lines.extend(generate_section('Usage', self.app.cfg['usage'], strip=True))
+
+        # Examples (optional)
+        lines.extend(generate_section('Examples', self.app.cfg['examples'], strip=True))
+
+        # Additional information: homepage + (if available) doc paths/urls, upstream/site contact
+        lines.extend(generate_section("More information", " - Homepage: %s" % self.app.cfg['homepage']))
+
+        docpaths = self.app.cfg['docpaths'] or []
+        docurls = self.app.cfg['docurls'] or []
+        if docpaths or docurls:
+            env_name = convert_name(self.app.name, upper=True)
+            root_envvar = ROOT_ENV_VAR_NAME_PREFIX + env_name
+            lines.extend([" - Documentation:"])
+            lines.extend(["    - $%s/%s" % (root_envvar,path) for path in docpaths])
+            lines.extend(["    - %s" % url for url in docurls])
+
+        upstream = self.app.cfg['upstream_contact']
+        if upstream:
+            if isinstance(upstream, list):
+                lines.extend([" - Upstream contacts:"])
+                lines.extend(["    - %s" % address for address in upstream])
+            else:
+                lines.extend([" - Upstream contact: %s" % upstream])
+
+        site_contact = self.app.cfg['site_contact']
+        if site_contact:
+            if isinstance(site_contact, list):
+                lines.extend([" - Site contacts:"])
+                lines.extend(["    - %s" % address for address in site_contact])
+            else:
+                lines.extend([" - Site contact: %s" % site_contact])
```

Comment:
since the code is copy-paste for `upstream_contacts` and `site_contacts`, how about:

```python
for contacts_type in ['upstream', 'site']:
    contacts = self.cfg['%s_contacts' % contacts_type]
    if contacts:
        if isinstance(contacts, list):
            lines.append(" - %s contacts" % contacts_type.capitalize())
            lines.extend(["    - %s" % c for c in contacts])
        else:
            lines.append(" - %s contact: %s" % (contacts_type.capitalize(), contacts))